classes/quiz_settings.php

```diff
@@ -208,6 +215,42 @@ protected function validate_allowedbrowserexamkeys($keys) {
         return true;
     }
 
+    /**
+     * Validate that if a file has been uploaded by current user, that it is a valid PLIST XML file.
+     *
+     * @param string $itemid Item ID of file in user draft file area.
+     * @return bool|lang_string
+     *
+     * @throws \coding_exception
+     * @throws IOException
+     */
+    protected function validate_sebconfigfile($itemid) {
+        // When saving the settings, this value will be null.
+        if (is_null($itemid)) {
+            return true;
+        }
+        // If there is a config file uploaded, make sure it is a PList XML file.
+        $file = settings_provider::get_current_user_draft_file($itemid);
```

Comment:
is it possible to move this out of the class as well? This is very couples with submitting the form which I'd like to avoid.